Sources/XcodeProj/Objects/Project/PBXProject.swift

```diff
@@ -154,20 +147,24 @@ public final class PBXProject: PBXObject {
     /// Local Swift packages.
     public var localPackages: [XCLocalSwiftPackageReference] {
         set {
-            var finalReferences: [PBXObjectReference] = packageReferences ?? []
-            let newReferences = newValue.references()
-            for reference in newReferences {
-                if !finalReferences.contains(reference) {
-                    finalReferences.append(reference)
-                }
-            }
-            packageReferences = finalReferences
+            setPackageReferences(newValue)
         }
         get {
             packageReferences?.objects() ?? []
         }
     }
 
+    private func setPackageReferences<T: PBXContainerItem>(_ references: [T]) {
```

Comment:
```suggestion
    private func setPackageReferences<T: PBXContainerItem>(_ packages: [T]) {
```

Reply:
This makes it more clear its a package reference and removes the `references.references()` issue